src/toil/jobStores/utils.py

```diff
@@ -109,3 +110,33 @@ def single_attempt():
             yield
 
         yield single_attempt()
+
+default_delays = (0, 1, 1, 4, 16, 64)
+default_timeout = 300
+
+
+def retry_http(delays=default_delays, timeout=default_timeout, predicate=None):
+    """
+    >>> i = 0
+    >>> for attempt in retry_http():
+    ...     with attempt:
+    ...         i += 1
+    ...         raise urllib2.HTTPError('http://www.test.com', '408', 'some message', {}, None)
+    Traceback (most recent call last):
+    ...
+    HTTPError: HTTP Error 408: some message
+    >>> print i
+    3
+    >>> for attempt in retry_http():
```

Comment:
Very good! I have a third one: changing any of the cases in a way that affects the expected value of `i` will require changing all cases below it.